packages/common/src/discussions/_internal/DiscussionSchema.ts

```diff
@@ -0,0 +1,49 @@
+import { IConfigurationSchema } from "../../core";
+import {
+  ENTITY_ACCESS_SCHEMA,
+  ENTITY_CATEGORIES_SCHEMA,
+  ENTITY_NAME_SCHEMA,
+  ENTITY_TAGS_SCHEMA,
+} from "../../core/schemas/shared";
+
+export type DiscussionEditorType = (typeof DiscussionEditorTypes)[number];
+export const DiscussionEditorTypes = ["hub:discussion:edit"] as const;
+
+/**
+ * defines the JSON schema for a Discussion's editable fields
+ */
+export const DiscussionSchema: IConfigurationSchema = {
+  required: ["title", "prompt"],
+  type: "object",
+  properties: {
+    title: ENTITY_NAME_SCHEMA,
+    prompt: {
```

Comment:
similarly, where is this stored on the backing item? Despite being labeled "Prompt", will this actually be stored on the item's `description` or `snippet`?

Reply:
Yes, you will need to determine where on the backing item you want this property to live, and then where to map it on your discussion entity. On projects, for example, we have a property `status` that is project-specific. It lives on the backing item's data (`item.data.status`) and then we [map](https://github.com/Esri/hub.js/blob/master/packages/common/src/projects/_internal/getPropertyMap.ts#L19) it to the top-level of the project entity.